components/timestamp/index.test.tsx

```diff
@@ -0,0 +1,120 @@
+    });
+
+    test('hourCycle should be h23 when military time is true and the prop was not set', () => {
+        const testState = {
+            ...initialState,
+            entities: {
+                ...initialState.entities,
+                preferences: {
+                    myPreferences: {
+                        'display_settings--use_military_time': 'true',
+                    },
+                },
+            },
+        } as unknown as GlobalState;
+
+        const props = mapStateToProps(testState, {});
+        expect(props.hourCycle).toBe('h23');
+    });
+
+    test('hourCycle should have the value of prop.hourCycle when given', () => {
+        const testState = {
+            ...initialState,
+            entities: {
+                ...initialState.entities,
+                preferences: {
+                    myPreferences: {
+                        'display_settings--use_military_time': 'true',
+                    },
+                },
+            },
+        } as unknown as GlobalState;
+
+        const props = mapStateToProps(testState, {hourCycle: 'h24'});
+        expect(props.hourCycle).toBe('h24');
+    });
+
+    test('timeZone should be the user TZ when the prop was not set', () => {
+        const testState = {
+            ...initialState,
+            entities: {
+                ...initialState.entities,
+                users: {
+                    ...initialState.entities.users,
+                    profiles: {
+                        ...initialState.entities.users.profiles,
+                        [currentUserId]: {
+                            ...initialState.entities.users.profiles[currentUserId],
+                            timezone: {
+                                useAutomaticTimezone: false,
+                                manualTimezone: 'Europe/Paris',
+                            },
+                        },
+                    },
+                },
+            },
+        } as unknown as GlobalState;
+
+        const props = mapStateToProps(testState, {});
+        expect(props.timeZone).toBe('Europe/Paris');
+    });
+
+    test('timeZone should be the value of prop.timeZone when given', () => {
+        const testState = {
+            ...initialState,
+            entities: {
+                ...initialState.entities,
+                users: {
+                    ...initialState.entities.users,
+                    profiles: {
+                        ...initialState.entities.users.profiles,
+                        [currentUserId]: {
+                            ...initialState.entities.users.profiles[currentUserId],
+                            timezone: {
+                                useAutomaticTimezone: false,
+                                manualTimezone: 'Europe/Paris',
+                            },
+                        },
+                    },
+                },
+            },
+        } as unknown as GlobalState;
```

Comment:
I think this might be a bit less verbose and easier to read when just modifying the one value. It would also naturally inherit its `GlobalState` type.

```suggestion
        const testState = {...initialState};
        testState.entities.users.profiles[currentUserId].timezone = {
            useAutomaticTimezone: false,
            manualTimezone: 'Europe/Paris',
        };
```

Reply:
Fixed, thanks!